enhancements/ingress/aws-load-balancer-operator.md

```diff
@@ -0,0 +1,394 @@
+provisioned have certain resource tags present on them. The operator can detect
+the subnets and tag them or the user can do this. This is specified with the
+_AutoTagSubnet_ field.
+
+The user can also specify the Ingress class which the lb-controller will
+reconcile. This could default to **_“alb”_**. This value is required because if
+an ingress class value is not passed to the controller it attempts to reconcile
+all Ingress resources which don’t have an ingress class annotation or where the
+ingress class field is empty. Passing the value of “alb” does not restrict the
+number of ingress classes which the controller can reconcile. This is described
+further in the [Parallel operation of Openshift
+Router](#parallel-operation-of-the-openshift-router-and-lb-controller) and
+lb-controller section.
+
+The _DeploymentConfig_ field contains any configuration required for the
+deployment. The number of replicas is the only field for now. But in future
+versions other fields from the
+[PodSpec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#podspec-v1-core)
+could also be exposed here.
+
+Finally, WAFs and DDos prevent configurations are attached to the load balancer
+through annotations specified on the Ingress resource. But users might want to
+limit which features are available and the `DisabledAddons` array can be used to
+specify which features/addons have to be disabled. By default all the addons
+will be enabled. They are disabled by setting the addon CLI flag to false. When
+an addon which was previously enabled is disabled the controller does not remove
+the existing addon attachment from the provisioned load balancers.
+
+The operator will also create a
+[CredentialRequest](https://docs.openshift.com/container-platform/4.9/rest_api/security_apis/credentialsrequest-cloudcredential-openshift-io-v1.html#credentialsrequest-cloudcredential-openshift-io-v1)
+on behalf of the lb-controller and mount the minted credentials in the
+_Deployment_ of the controller. This means that the operator will only work on
+OCP/OKD but this is sufficient for the initial release.
+
+
+### Risks and Mitigations
+
+#### Restricting features in the lb-controller
+
+Some changes will have to be made in the lb-controller to ensure that the users
+only use features which are supported on OCP and only those features which
+target Ingress resources. These changes may also be contributed back upstream
+since it may benefit other users.
+
+1. The annotation `alb.ingress.kubernetes.io/target-type` can be set to the
+value **_ip_** or **_instance_**. When the value is set to _instance_ the
+created load balancer will be configured to route traffic through a NodePort
+service. The _ip_ value only works when the cluster uses the Amazon VPC CNI and
+hence will not work on OCP/OKD clusters. The controller has to be modified so
+that this annotation is ignored and the controller always uses routing through
+_NodePort_ services.
```

Comment:
Does this affect source preservation? 

Maybe worth mentioning <https://bugzilla.redhat.com/show_bug.cgi?id=2023681>.  Based on <https://bugzilla.redhat.com/show_bug.cgi?id=2023681#c2>, I think you will be unaffected if you use `ip`.

Reply:
The bugs you mention are for NLBs but the Ingresses will always use ALBs which don't preserve the source.